src/vehicle_part.cpp

```diff
@@ -486,6 +486,11 @@ bool vehicle_part::is_battery() const
     return base.is_magazine() && base.ammo_types().count( ammotype( "battery" ) );
 }
 
+bool vehicle_part::is_fuelbunker() const
+{
+    return base.is_magazine() && base.ammo_types().count( ammotype( "charcoal" ) );
```

Comment:
No.

It's bad enough that we're special casing batteries, but this code was written to support arbitrary solid fuel and needs to continue supporting arbitrary solid fuel.  Figure something else out.